lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1285,17 +1211,80 @@ ConstExprFunctionState::getSingleWriterAddressValue(SILValue addr) {
 
     // If this is some other user that we don't know about, then we should
     // treat it conservatively, because it could store into the address.
-    return evaluator.getUnknown(addr, UnknownReason::Default);
+    return {false, evaluator.getUnknown(addr, UnknownReason::Default)};
   }
 
-  // If we found a store of a constant, then return that value!
-  if (memoryObject->getValue().isConstant()) {
-    setValue(addr, memoryAddress);
-    return memoryAddress;
+  if (mustCheckAggregateInitialized && !checkAggregateInitialized())
+    return {false, evaluator.getUnknown(addr, UnknownReason::Default)};
+
+  return {initialized, None};
```

Comment:
do we want to assert that if `initialized` is true, then `memoryObject->getValue().isConstant()`?

alternatively, would it be simpler to remove `initialized` altogether? seems we can just return a llvm::Optional<SymbolicValue> instead of a struct:
1. if the value is None, we have no writer
2. if it's a const, a single writer has been found and processed
3. otherwise, it must be Unknown (unsupported inst types or multiple writers found) #

Reply:
Ah yes, I can remove `initialized`. I'll make it an optional symbolicvalue, slightly different from your suggestion:

1. if there are no writers -> None
2. if a writer successfully initializes it -> None
3. otherwise -> Unknown

This is more consistent with things like `computeCallResult` that write values indirectly. Also, it's simpler to implement than something that returns the actual initialized value. And no callers actually need to know what the actual initialized value is.

Reply:
SG! I assume we don't want to return a bool, because the unknown value could have some error msg context?